### PR TITLE
Add LLaMA C++ models and Fix PyType warning/errors raised in VS Code

### PR DESCRIPTION
**Pull request by D-X-Y**
- Add a LLaMA C++ model at `langfun/core/llms/llama_cpp.py`
- Fix various minor pytype or annotation issues.

### CODE REVIEW

File: langfun/core/llms/__init__.py [hunk 1]
```diff
@@ -45,6 +45,9 @@
 from langfun.core.llms.openai import Gpt3Babbage
 from langfun.core.llms.openai import Gpt3Ada
 
+# LLaMA C++ models.
+from langfun.core.llms.llama_cpp import RemoteLLaMACPP
```
Comment (daiyip) [inline langfun/core/llms/__init__.py:49]: Consider `LlamaCppRemote`.

File: langfun/core/llms/llama_cpp.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  The Remote LLaMA C++ models can be launched via
+  https://github.com/ggerganov/llama.cpp/tree/master/examples/server
+  """
+
+  url: Annotated[
+      str | None,
+      'The name of the model to use.',
+  ] = None
+
+  name: Annotated[
```
Comment (daiyip) [inline langfun/core/llms/llama_cpp.py:35]: Is the model identified by the url alone? Thus this is not needed?

File: langfun/core/llms/llama_cpp.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+      results = []
+      for prompt in cur_prompts:
+        result = lf.LMSamplingResult()
+        for _ in range(self.sampling_options.n or 1):
+          data = {
+                "prompt": prompt.text,
+                "n_predict": self.sampling_options.max_tokens,
+                "temperature": self.sampling_options.temperature,
+                "top_k": self.sampling_options.top_k or 50,
+                "top_p": self.sampling_options.top_p or 0.95,
```
Comment (daiyip) [inline langfun/core/llms/llama_cpp.py:61]: Consider setting the default sampling_options of this class via:
```
sample_options = lf.LMSamplingOptions(top_k=50, top_p=0.95)
```

File: langfun/core/llms/llama_cpp.py
Comment (daiyip) [inline langfun/core/llms/llama_cpp.py:1]: Consider to add a test for this file? I know we will have to mock, it's to protect the class from regression by future changes.

**Review by daiyip — CHANGES_REQUESTED** [review]
Thanks for adding LLamaCpp models, Xuanyi! I have put a few comments inline.

File: langfun/core/llms/__init__.py (hunk 1 above)
Comment (D-X-Y) [reply langfun/core/llms/__init__.py]: Done

File: langfun/core/llms/llama_cpp.py (hunk 2 above)
Comment (D-X-Y) [reply langfun/core/llms/llama_cpp.py:35]: Any user can launch any kind of model with whatever url as he wants. The name is not associated with the url, I view it as a model name to indicate the real model behind the url for the convenience purpose. WDYT?

File: langfun/core/llms/llama_cpp.py (hunk 3 above)
Comment (D-X-Y) [reply langfun/core/llms/llama_cpp.py:61]: This is more used to avoid people input a `None` value to top_k?

File: langfun/core/llms/llama_cpp.py
Comment (D-X-Y) [reply langfun/core/llms/llama_cpp.py:1]: Done~

File: langfun/core/llms/llama_cpp.py (hunk 2 above)
Comment (daiyip) [reply langfun/core/llms/llama_cpp.py:35]: It makes sense. Thanks Xuanyi.

**Review by daiyip — APPROVED** [review]

**Pull request merged**